### DISCUSSION TITLE
Pinning repos

### DISCUSSION THREAD

**Discussion by sfullerbeckman** [community/community#4332] [Other Feature Feedback, Questions, & Ideas]
# Pinning Repos

Github has been making it easier to explore all the repos across the Github ecosystem. They have even added recommended repos based on users' interests. But one thing that Github.com does not support yet is that ability for a user to "pin" repos that he/she is interested in or often reference.

There are many possibilities for how this could look and function in the UI, but it would greatly improve the Github.com interface if there was the ability to pin repositories. Here are a couple of the benefits of such a feature:
- Quickly access repos that he/she don't have time to look at the moment but want to go back to later
- Bookmark repos that he/she access often

**Update:** As it seems that some people are interested in this feature, please be sure to up the vote of this post and/or put a thumbs up so that Github staff/devs will take notice of this discussion.

**Comment by tallguyjenks** [comment]
Are you using the [⭐ Star] feature on Repo's for some other purpose? I was always under the impression that the [⭐ Star] feature is what serves the purpose you're describing?

**Reply by sfullerbeckman** [reply]
I personally don't like using the Star feature for pinning. I view the Star feature as more of a way to tell the repo's maintainer that I like his or her repo (a method of positive feedback), not as a way of pinning a repo for my own reference.

**Reply by Levi-Lesches** [reply]
Maybe think of the ⭐ as exactly what you're asking for -- a way to keep track of the repo. I see the positive feedback as implicit; it makes me happy to know that people are pinning my repo.

**Reply by sfullerbeckman** [reply]
The star feature still doesn't accomplish what I want. For example, what if I wanted to "bookmark"  a few repositories that my team has been working on. I wouldn't necessarily want to star those. I wouldn't necessarily want to give a positive star. Another example, what if, like I wrote in my original post, I wanted to bookmark/pin a repository that I want to look at in depth later? I wouldn't necessarily want to Star this, especially since, after I look at in depth at a later point, I might deem it to be a poor repository not worthy of a star. 

The Star carries a positive connotation. Thus, whenever it is used, positive feedback is being given. I want a way to pin repositories whether they are good, bad, or unknown. I want to pin for the sake of being able to quickly reference repositories at any point.

**Reply by Levi-Lesches** [reply]
> I wouldn't necessarily want to Star this, especially since, after I look at in depth at a later point, I might deem it to be a poor repository not worthy of a star.

...then you remove the star?

> The Star carries a positive connotation. Thus, whenever it is used, positive feedback is being given.

To you, sure. But it's not _the only_ interpretation of the star. From the [docs](https://docs.github.com/en/get-started/exploring-projects-on-github/saving-repositories-with-stars): 

> Starring makes it easy to find a repository or topic again later. You can see all the repositories and topics you have starred by going to your stars page.
> 
> You can star repositories and topics to discover similar projects on GitHub. When you star repositories or topics, GitHub may recommend related content in the discovery view of your news feed. For more information, see "Finding ways to contribute to open source on GitHub".
> 
> Starring a repository also shows appreciation to the repository maintainer for their work. Many of GitHub's repository rankings depend on the number of stars a repository has. In addition, Explore shows popular repositories based on the number of stars they have.

The point is, use stars to "pin" repositories, and GitHub will implicitly learn that "hey, this caught you're attention, that's probably a good thing". But you're not _legally obligated_ to like what you see.

**Reply by sfullerbeckman** [reply]
I see what you are saying, @Levi-Lesches. And, yes, people can use stars like this if they so choose. But I think that Github can at least improve on the whole "pinning" concept. If they use a star, they are triggering Github to recommend related content as well as giving positive feedback to the repo, which both may be something that they don't want to do. I still think that there is benefit to incorporating "pinning" functionality to Github.

If anyone else has any thoughts related to this, feel free to chime in. I specifically would like to hear from any Github staff (@aconsuegra / @michellemerrill / @evi-liu) if possible.

**Reply by tallguyjenks** [reply]
I didn't know about the related content suggestion functionality, with that new insight, I'm in agreement with @sfullerbeckman, that would be akin to liking YouTube videos and getting suggested videos v.s. adding a few select videos to a watch later playlist for later. I understand the distinction now

**Reply by hayesall** [reply]
To summarize: A [ Star :star: ] means different things to different people. Nadia Eghbal suggested they're used for both '**bookmarking/favoriting**' and '**liking**' in "[Roads and Bridges: The Unseen Labor Behind our Digital Infrastructure](https://www.fordfoundation.org/media/2976/roads-and-bridges-the-unseen-labor-behind-our-digital-infrastructure.pdf#page=130)," (e.g. see the top of page 130)—I think she mentioned something similar in "Working in Public," but I haven't found it on short notice.

I use stars for both purposes.

**Reply by cryptictech** [reply]
I agree with the notion that the star 🌟 is used as more of a Facebook like than a bookmark and that it would be nice to be able to bookmark and organize repositories that we use on a more daily basis separately from the star system.

I think it is also correct that it means different things to different people and that people who use it as a bookmark could also benefit from having a dedicated system to the issue as it would free them up to "like" or star less relevant but still good projects.

**Comment by Zerotask** [comment]
That's a nice feature which is already implemented in the android app (Favorites), but missing  on the web.

I also suggested this on the GitHub support forum.
I hope it will get more attraction here

**Reply by sfullerbeckman** [reply]
Agreed. I believe it is implemented on the iPhone version as well, but I believe most people use Github from the browser far more than mobile devices. So, it would be very helpful if it could be integrated into the web platform.

**Comment by hayesall** [comment]
I think there are two requests here: *pinning* and *better sorting*.

For *sorting*, it might make sense to:

1. Store the "*number of times you've accessed a repo*" in a [priority queue](https://en.wikipedia.org/wiki/Priority_queue)
2. Then: "*Sort by: Number of visits*" in https://github.com/stars

**Reply by sfullerbeckman** [reply]
Can you explain what you are referring to as far as sorting? Are you saying that you would like there to be additional ways to sort the repo listing of all those that you have given stars?

**Reply by hayesall** [reply]
> you would like there to be additional ways to sort the repo listing

Pretty much. Currently the star list can be sorted three ways—but "*Recently Starred*" is the only one I find helpful.

![Screenshot from a GitHub Stars Page. The three sort options are: Recently starred, Recently active, and Most stars](https://user-images.githubusercontent.com/11916674/124024450-35d30400-d9bd-11eb-9e5f-e7f595cd4507.png)

I brought this up because GitHub tracks access statistics for each repository (see below), so it seems possible to use this for sorting (e.g.: "How often do **I** look at each repository I have bookmarked?").

Here's an example from a 'Traffic' tab showing views and unique visitors:

![image](https://user-images.githubusercontent.com/11916674/124027233-a2033700-d9c0-11eb-8ebc-b15befe7dd17.png)

**Reply by sfullerbeckman** [reply]
I like the idea. As this feature feels separate from the topic of pinning repositories, would you create a different feature request/discussions thread for this? We can carry out additional discussion on there. Thank you.

**Comment by sfullerbeckman** [comment]
A couple ideas:
- Maybe there could be a dropdown button called `Pins` in the header title bar that, when clicked, displays a dropdown of all of the repos that you've pinned. This could sit between the `Issues` and `Marketplace` buttons
- Or to the left of the notification icon in the header title bar there could be a pin icon (![baseline_push_pin_black_24dp](https://user-images.githubusercontent.com/83652131/124164953-493ea780-da6f-11eb-8e40-30e47d2f58d9.png)) which is also a dropdown where you can see all of the repos you have pinned.

Those are just a couple of ideas.

**Comment by sfullerbeckman** [comment]
Anyone know if GitHub supports this idea? I've received no word whether this is a feature they want to implement or not. I've not received word from them on this other feature request I made either... #16796

@davidstaheli, do you have any feedback on this?
